src/Microsoft.OData.Core/UriParser/ODataQueryOptionParser.cs

```diff
@@ -326,6 +331,31 @@ public string ParseDeltaToken()
             string deltaTokenQuery;
             return this.TryGetQueryOption(UriQueryConstants.DeltaTokenQueryOption, out deltaTokenQuery) ? deltaTokenQuery : null;
         }
+
+        /// <summary>
+        /// Parses a compute clause on the given full Uri, binding
+        /// the text into semantic nodes using the constructed mode.
+        /// </summary>
+        /// <returns>A <see cref="ComputeClause"/> representing the computed properties.</returns>
+        public ComputeClause ParseCompute()
+        {
+            if (this.computeClause != null)
+            {
+                return this.computeClause;
+            }
+
+            string computeQuery;
+
+            if (!this.TryGetQueryOption(UriQueryConstants.ComputeQueryOption, out computeQuery)
+                || string.IsNullOrEmpty(computeQuery)
+                || this.targetEdmType == null)

```

Comment:
>|| this.targetEdmType == null [](start = 16, length = 29)

Under what situations will targetEdmType be null (a dynamic property?) and what are the ramifications of returning a null in this case? Is the compute clause silently ignored? Do we have a test case for this case?